src/plugins/workspace/server/saved_objects/workspace_id_consumer_wrapper.ts

```diff
@@ -126,8 +156,60 @@ export class WorkspaceIdConsumerWrapper {
         }
         return wrapperOptions.client.find(finalOptions);
       },
-      bulkGet: wrapperOptions.client.bulkGet,
-      get: wrapperOptions.client.get,
+      bulkGet: async <T = unknown>(
+        objects: SavedObjectsBulkGetObject[] = [],
+        options: SavedObjectsBaseOptions = {}
+      ): Promise<SavedObjectsBulkResponse<T>> => {
+        const { workspaces } = this.formatWorkspaceIdParams(wrapperOptions.request, options);
+        if (!!workspaces && workspaces.length > 1) {
+          // Version 2.18 does not support the passing of multiple workspaces.
+          throw SavedObjectsErrorHelpers.createBadRequestError('Multiple workspace parameters');
+        }
+
+        const objectToBulkGet = await wrapperOptions.client.bulkGet<T>(objects, options);
+
+        if (workspaces?.length === 1) {
+          return {
+            ...objectToBulkGet,
+            saved_objects: objectToBulkGet.saved_objects.map((object) => {
+              return this.validateObjectInAWorkspace(object, workspaces[0], wrapperOptions.request)
+                ? object
+                : {
+                    ...object,
+                    error: {
+                      ...SavedObjectsErrorHelpers.createGenericNotFoundError(object.type, object.id)
+                        .output.payload,
+                    },
+                  };
+            }),
+          };
+        }
+
+        return objectToBulkGet;
+      },
+      get: async <T = unknown>(
+        type: string,
+        id: string,
+        options: SavedObjectsBaseOptions = {}
+      ): Promise<SavedObject<T>> => {
+        const { workspaces } = this.formatWorkspaceIdParams(wrapperOptions.request, options);
+        if (!!workspaces && workspaces.length > 1) {
+          // Version 2.18 does not support the passing of multiple workspaces.
+          throw SavedObjectsErrorHelpers.createBadRequestError('Multiple workspace parameters');
+        }
+
+        const objectToGet = await wrapperOptions.client.get<T>(type, id, options);
+
+        if (
+          workspaces?.length === 1 &&
+          !this.validateObjectInAWorkspace(objectToGet, workspaces[0], wrapperOptions.request)
+        ) {
+          throw SavedObjectsErrorHelpers.createGenericNotFoundError(type, id);
```

Comment:
As discussed we could use 403 forbidden here to differentiate object not fould case.

Reply:
Sure I will update the code.